pages/knowledge/colours-and-contrast/text/README.md

```diff
@@ -0,0 +1,79 @@
+
+To illustrate this: the following image shows the full spectrum of possible contrast ratios from `1:1` on the left to `21:1` on the right.
+
+![Text with colour gradient overlay](_media/lazy-dog.png)
+
+Somewhere around the “x” the contrast ratio reaches `3:1`. Somewhere around the “m” we arrive at `4.5:1`. 
+
+For a more direct comparison: the following image shows both large and default text with contrast ratios of exactly `3:1` and `4.5:1`.
+
+![Small and large text](_media/small-and-large-text.png)
+
+It may not seem like much of a difference, but in fact it heavily enlargens the number of possible colour combinations.
+
+It is important to mention that the values `3:1` and `4.5:1` represent a *minimal* standard for *acceptable* contrast, they don’t represent an ideal. They should be treated as threshold and not as target; aiming higher is recommended. 
+
+## Enhancing text contrast
+
+Besides changing the colour of the text or the background there are many other ways do enhance contrast. For example:
+
+### Adding shadow
+
+Drop shadow around text can make up for insufficient contrast between text colour and background.
+
+For example, look at the following image. It shows two words, both with a red fill that has an insufficient contrast of `2.1:1` to the white background.
+
+![A word without and a word with shadow](_media/words-without-and-with-shadow.png)
+
+The shadow of the second word results in a contrast of appropriately `4.5:1`.
+
+### Adding a contrast layer
+
+In some cases, text is displayed on non-homogeneous backgrounds. Depending on the heterogeneity of the background, the text may be very hard to read.
+
+For example, look at the following image. It shows a white text “Welcome to the beach!” on blue sky with some white clouds. While there is decent contrast between white text and blue sky there is almost no contrast between white text and white clouds. Overall it’s very hard to read.
+
+![White text on blue sky and white clouds](_media/beach.png)
+
+You could increase contrast by adding a text shadow as described above. Or you could add a semi-transparent darkening layer between text and background
+
+![White text with semi-transparent darkening background](_media/beach-with-background.png)
+
+Of course you can also combine different solutions to add up their effects.
+
+## Exceptions
+
+There are some exceptions to the contrast requirements of text content.
+
+- **Logos** usually must adhere strictly to corporate design guidelines and thus may be displayed in those colours regardless of contrast ratios (needless to say that it still is unpleasant if your clients cannot perceive your logo).
+- **Decorative text:** this could be the background pattern of an image that is created from random words.
+- **Incidental text:** this could be a random street sign somewhere in the background of a photograph.
+- **Disabled text:** this could be the label of a disabled element in an online form.
+
```

Comment:
We should at least mention `placeholder` values here. They are a really nasty edge case, and they are talked about a lot already in other places (like https://www.nngroup.com/articles/form-design-placeholders/).

Reply:
Standpoint A4A
* Mustn’t replace the label
* *if* it does it has to be like in Material Design (leaving the field on focus)
* For an *actual* placeholder too much contrast can confuse people (did I enter this?)
* Could also be in “forms” section.

**To Do:** Check https://www.nngroup.com/articles/form-design-placeholders/ and mention placeholder as they describe it.

Reply:
I checked the WCAG page [Understanging Success Criterion 1.4.3: Contrast (Minimum)](https://www.w3.org/WAI/WCAG21/Understanding/contrast-minimum.html). There they state explicitly “The minimum contrast success criterion (1.4.3) applies to text in the page, including placeholder text and text that is shown when […]”

I added a sub title and short paragraph to the “Exceptions” section. I state what the specification says while also mentioning the general usability issues. For more information I link to the NNGroup article.

> What about placeholder text?
> 
> The WCAG's contrast requirement for text also applies to placeholder text. But it's important to mention that placeholders offers a bunch of usability and accessibility challenges beyond just adequate contrast – sometimes even because of adequate contrast. For more information see this [article about why placeholders in form fields are harmful](https://www.nngroup.com/articles/form-design-placeholders/)